### PR TITLE
CSP-66-outer-ProvideCourseLocations

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.Roatp.CourseManagement.Api.UnitTests/Controllers/ProviderCourseControllerTests.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class ProviderCourseControllerTests
         public async Task GetProviderCourse_ReturnsExpectedState(int ukprn,int larsCode, int expectedStatusCode)
         {
             var mediatorMock = new Mock<IMediator>();
-            mediatorMock.Setup(m => m.Send(It.Is<GetProviderCourseQuery>(q => q.LarsCode == larsCode && q.Ukprn==ukprn), It.IsAny<CancellationToken>())).ReturnsAsync(new GetProviderCourseResult { LarsCode = ValidLarsCode});
+            mediatorMock.Setup(m => m.Send(It.Is<GetProviderCourseQuery>(q => q.LarsCode == larsCode && q.Ukprn==ukprn ), It.IsAny<CancellationToken>())).ReturnsAsync(new GetProviderCourseResult { LarsCode = ValidLarsCode});
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Api.UnitTests/Controllers/ProviderCourseControllerTests.cs:28]: undo the change, not needed

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourse/GetProviderCourseQueryHandler.cs [hunk 2]
```diff
@@ -41,6 +43,27 @@ public async Task<GetProviderCourseResult> Handle(GetProviderCourse.GetProviderC
                 return null;
             }
 
+            var providerCourseLocations = await _courseManagementApiClient.Get<List<GetProviderCourseLocationsResponse>>(new GetProviderCourseLocationsRequest(request.Ukprn, request.LarsCode));
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourse/GetProviderCourseQueryHandler.cs:46]: We're trying to use a different approach with calling information from the inner apis.   We're using GetWithResponseCode, and if there is we get a response code we don't expect, we just pass the response code back.r
I will double-check with Deven, and will reply here about it

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourse/GetProviderCourseQueryHandler.cs [hunk 3]
```diff
@@ -41,6 +43,27 @@ public async Task<GetProviderCourseResult> Handle(GetProviderCourse.GetProviderC
...
+            List<ProviderCourseLocationModel> locations = new List<ProviderCourseLocationModel>();
+            foreach (var location in providerCourseLocations)
+            {
+                var providerCourseLocationModel = new ProviderCourseLocationModel
+                {
+                    LocationName = location.LocationName,
+                    LocationType = location.LocationType,
+                    HasBlockReleaseDeliveryOption = location.HasBlockReleaseDeliveryOption,
+                    HasDayReleaseDeliveryOption = location.HasDayReleaseDeliveryOption,
+                    OffersPortableFlexiJob = location.OffersPortableFlexiJob,
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourse/GetProviderCourseQueryHandler.cs:62]: you can remove portable flexijob

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourse/GetProviderCourseQueryHandler.cs [hunk 4]
```diff
@@ -41,6 +43,27 @@ public async Task<GetProviderCourseResult> Handle(GetProviderCourse.GetProviderC
...
+            if (!providerCourseLocations.Any())
+            {
+                _logger.LogError($"Provider course locations not found for ukprn: {request.Ukprn} ProviderCourseId: {course.ProviderCourseId}");
+                return null;
+            }
+
+            List<ProviderCourseLocationModel> locations = new List<ProviderCourseLocationModel>();
+            foreach (var location in providerCourseLocations)
+            {
+                var providerCourseLocationModel = new ProviderCourseLocationModel
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourse/GetProviderCourseQueryHandler.cs:56]: use an implicit mapper here?

**Review by MarkFCain — COMMENTED** [review]
some comments that can be addressed, but leave the issue raised in the handler till I respond

File: src/SFA.DAS.Roatp.CourseManagement.Api/Controllers/ProviderCourseController.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@
 namespace SFA.DAS.Roatp.CourseManagement.Api.Controllers
 {
     [ApiController]
-    [Route("[controller]/")]
+    //[Route("[controller]/")]
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Api/Controllers/ProviderCourseController.cs]: Delete commented code

**Review by DevenKShah — APPROVED** [review]

**Review by MarkFCain — DISMISSED** [review]

**Review by MarkFCain — APPROVED** [review]
